### PR TITLE
Acquire `da_finalization_period` from the command line

### PR DESCRIPTION
**Pull request by rafal-ch**
## Linked Issues
Closes https://github.com/FuelLabs/fuel-core/issues/2218

## Description
This PR extends the set of accepted command line parameters. The user is now able to provide the DA update time via the new `--da-finalization-period` argument.

It also fixes the issue with `unrecorded_blocks` collection which was not saturated with blocks.

## Checklist
- [X] New behavior is reflected in tests

### Before requesting review
- [X] I have reviewed the code myself

### CODE REVIEW

File: crates/fuel-gas-price-algorithm/gas-price-analysis/src/main.rs [hunk 1]
```diff
@@ -37,6 +37,9 @@ struct Arg {
     /// File path to save the chart to. Will not generate chart if left blank
     #[arg(short, long)]
     file_path: Option<String>,
+    #[arg(short, long)]
+    /// DA finalization period in blocks
```
Comment (MitchTurner) [inline crates/fuel-gas-price-algorithm/gas-price-analysis/src/main.rs:41]:
```suggestion
    /// DA finalization period in L2 blocks
```

File: crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs [hunk 2]
```diff
@@ -177,11 +181,13 @@ impl Simulator {
...
     }
 
-    fn zip_l2_blocks_with_da_blocks(
+    fn calculate_da_blocks(
         &self,
         da_recording_rate: usize,
+        da_finalization_rate: usize,
         fullness_and_bytes: &Vec<(u64, u64)>,
     ) -> Vec<Option<Vec<RecordedBlock>>> {
+        let nones = std::iter::repeat(None).take(da_finalization_rate);
```
Comment (MitchTurner) [inline crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs:190]:
```suggestion
        let l2_blocks_with_no_da_blocks = std::iter::repeat(None).take(da_finalization_rate);
```

File: crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs [hunk 3]
```diff
@@ -207,7 +213,7 @@ impl Simulator {
                     }
                 },
             );
-        da_blocks
+        nones.chain(da_blocks).collect()
```
Comment (MitchTurner) [inline crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs:216]:
```suggestion
        l2_blocks_with_no_da_blocks.chain(da_blocks).collect()
```

File: crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs (hunk 2 above)
Comment (rafal-ch) [reply crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs]: Updated in https://github.com/FuelLabs/fuel-core/pull/2240/commits/ff0db6e70e8174fceb5484d13bb9e5ae9a917f68

File: crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs (hunk 3 above)
Comment (rafal-ch) [reply crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs]: Updated in https://github.com/FuelLabs/fuel-core/pull/2240/commits/ff0db6e70e8174fceb5484d13bb9e5ae9a917f68

**Review by MitchTurner — APPROVED** [review]

File: crates/fuel-gas-price-algorithm/src/v1.rs [hunk 4]
```diff
@@ -239,6 +239,12 @@ impl AlgorithmUpdaterV1 {
             // gas prices
             self.update_exec_gas_price(used, capacity);
             self.update_da_gas_price();
+
+            // metadata
+            self.unrecorded_blocks.push(BlockBytes {
+                height,
+                block_bytes,
+            });
```
Comment (xgreenx) [inline crates/fuel-gas-price-algorithm/src/v1.rs:247]: I don't see how it is related to the change in the gas price analyser

Comment (rafal-ch) [reply crates/fuel-gas-price-algorithm/src/v1.rs:247]: True, it's not related. It was something I discovered when working on this issue and decided to add a fix in the same PR.

Even though the PR description mentions this change, I should probably issue a separate ticket and create a different PR.

Would you like me to clean-up the situation for this case or you think we can merge as is?

**Review by xgreenx — APPROVED** [review]

File: crates/fuel-gas-price-algorithm/src/v1.rs (hunk 4 above)
Comment (xgreenx) [reply crates/fuel-gas-price-algorithm/src/v1.rs:247]: If @MitchTurner is okay of merging it in this PR, I'm also okay=)

**Review by xgreenx — APPROVED** [review]

**Pull request merged**